### PR TITLE
[Jetsurvey] fixed app crash on changing date selection

### PR DESCRIPTION
**Pull request by Abhimanyu14**
Fixed the app crash on changing the selected date for the date selection question.

Issue link: https://github.com/android/compose-samples/issues/512

### CODE REVIEW

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt [hunk 1]
```diff
@@ -95,7 +98,11 @@ class SurveyFragment : Fragment() {
         activity?.let {
             picker.show(it.supportFragmentManager, picker.toString())
             picker.addOnPositiveButtonClickListener {
-                viewModel.onDatePicked(questionId, picker.headerText)
+                val selectedData = Date().apply {
+                    time = picker.selection ?: date
+                }
+                val dateFormatted = SimpleDateFormat("EEE, MMM d", Locale.getDefault()).format(selectedData)
+                viewModel.onDatePicked(questionId, dateFormatted)
```
Comment (florina-muntenescu) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt:105]: With this implementation, you're putting a lot of logic in the UI that can't be unit tested. Move this to the ViewModel.

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt [hunk 2]
```diff
@@ -119,7 +119,7 @@ class SurveyViewModel(
                 }
             val answer: Answer.Action? = question.answer as Answer.Action?
             if (answer != null && answer.result is SurveyActionResult.Date) {
-                val formatter = SimpleDateFormat("MMM dd, yyyy", Locale.ENGLISH)
+                val formatter = SimpleDateFormat("EEE, MMM d", Locale.ENGLISH)
```
Comment (florina-muntenescu) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt:122]: The data format pattern seems to be used multiple times, move it to a constant

**Review by florina-muntenescu — CHANGES_REQUESTED** [review]

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt (hunk 1 above)
Comment (Abhimanyu14) [reply Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt]: moved the logic into view model

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt (hunk 2 above)
Comment (Abhimanyu14) [reply Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt]: created a const for the simple date format pattern

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt [hunk 3]
```diff
@@ -28,6 +28,9 @@ import androidx.fragment.app.viewModels
 import com.example.compose.jetsurvey.R
 import com.example.compose.jetsurvey.theme.JetsurveyTheme
 import com.google.android.material.datepicker.MaterialDatePicker
+import java.text.SimpleDateFormat
+import java.util.Date
```
Comment (florina-muntenescu) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt:32]: these imports are no longer needed.
make sure you run `spottlessApply`

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt [hunk 4]
```diff
@@ -68,8 +70,12 @@ class SurveyViewModel(
         _uiState.value = SurveyState.Result(surveyQuestions.surveyTitle, result)
     }
 
-    fun onDatePicked(questionId: Int, date: String) {
-        updateStateWithActionResult(questionId, SurveyActionResult.Date(date))
+    fun onDatePicked(questionId: Int, pickerSelection :Long?) {
+        val selectedData = Date().apply {
```
Comment (florina-muntenescu) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt:74]: typo - `selectedDate`

**Review by florina-muntenescu — CHANGES_REQUESTED** [review]
couple of more small changes

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt (hunk 4 above)
Comment (Abhimanyu14) [reply Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt]: fixed

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt (hunk 3 above)
Comment (Abhimanyu14) [reply Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt]: fixed

**Comment by Abhimanyu14** [comment]
@florina-muntenescu ,
Any more change required?

**Review by florina-muntenescu — APPROVED** [review]

**Comment by florina-muntenescu** [comment]
I missed this. Thank you @Abhimanyu14 for the changes!

**Pull request merged**